Database/messageRepository.cpp

```diff
@@ -50,8 +50,62 @@ void MessageRepository::Insert(const Message& message)
     std::string messageFilePath = databaseFolder + "/" + recipient + "/" + std::to_string(messageId) + ".txt";
 
     if(fileManager.Exists(messageFilePath))
-        throw new MessageRepositoryException("Error inserting Message ID: " + std::to_string(messageId) + ". File already exists");
+        throw MessageRepositoryException("Error inserting Message ID: " + std::to_string(messageId) + ". File already exists");
     
     std::cout << "Creating MessageEntry: " << messageFilePath << std::endl;
     fileManager.writeToFile(messageFilePath, message.ToString());
+}
+
+std::vector< std::string > MessageRepository::GetMessages(const std::string& username)
+{
+    std::lock_guard<std::mutex> guard(this->fileAccessMutex);
+    std::vector<std::string> messages;
+
+    if(fileManager.Exists(databaseFolder + "/" + username))
+    {
+        auto files = fileManager.getFiles( databaseFolder + "/" + username, true );        
+        messages.reserve(files.size());
+
+        for(auto path : files)
+            messages.emplace_back( fileManager.readFile( databaseFolder + "/" + username + "/" + path) );
+    }
+
+    return messages;
+}
+
+std::string MessageRepository::GetMessage(const std::string& username, int index)
+{
+    std::lock_guard<std::mutex> guard(this->fileAccessMutex);
+    
+    if(!fileManager.Exists( databaseFolder + "/" + username))
+        throw MessageRepositoryException("user hast no mailbox");
+
+    auto files = fileManager.getFiles( databaseFolder + "/" + username, true );
+    std::string message;
+
+    try {
+        message = fileManager.readFile( databaseFolder + "/" + username + "/" + files.at(index) );
+    } catch(...) {
+        throw MessageRepositoryException("message does not exist");
+    }
+
+    return message;
+}
+
+void MessageRepository::DeleteMessage(const std::string& username, int index)
+{
+    std::lock_guard<std::mutex> guard(this->fileAccessMutex);
+
+    if(!fileManager.Exists( databaseFolder + "/" + username))
+        throw MessageRepositoryException("user hast no mailbox");
+
+    auto files = fileManager.getFiles( databaseFolder + "/" + username, true );
+
+    try {
+        fileManager.deleteFile( databaseFolder + "/" + username + "/" + files.at(index) );
+    } catch(...) {
+        throw MessageRepositoryException("file not found for removal");
```

Comment:
Passiert das wirklich nur wenn das file nicht existiert? Wenn ja brauchen wir das möglicherweise nicht zu behandeln weil in beiden Fällen das gewünschte Ergebnis erreicht wurde.

Reply:
ich verstehe nicht ganz was du da meinst :thinking: